neurom/core/tests/test_iter.py

```diff
@@ -87,41 +99,46 @@ def test_iter_sections_default():
                           [n for n in core.iter_sections(POP)])
 
 
+def test_iter_sections_inrn():
+    assert_sequence_equal(list(core.iter_sections(NRN1, iterator_type=Tree.ipreorder)),
+                          list(core.iter_sections(NRN1, iterator_type=Tree.inrnorder)))
+
+
 def test_iter_sections_filter():
 
     for ntyp in nm.NEURITE_TYPES:
         a = [s for n in filter(lambda nn: nn.type == ntyp, POP.neurites)
              for s in n.iter_sections()]
-        b = [n for n in core.iter_sections(POP, neurite_filter=lambda n : n.type == ntyp)]
+        b = [n for n in core.iter_sections(POP, neurite_filter=lambda n: n.type == ntyp)]
         assert_sequence_equal(a, b)
 
+def test_iter_sections_inrnorder():
+    assert_sequence_equal([s.id for n in POP.neurites for s in n.iter_sections(Tree.inrnorder)],
+                          [2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 2, 3, 4])
 
-def test_iter_sections_ipostorder():
+def test_iter_sections_ipreorder():
+    assert_sequence_equal([s.id for n in POP.neurites for s in n.iter_sections(Tree.ipreorder)],
+                          [2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 2, 3, 4])
 
-    ref = [s for n in POP.neurites for s in n.iter_sections(Tree.ipostorder)]
-    assert_sequence_equal(ref,
-                          [n for n in core.iter_sections(POP, iterator_type=Tree.ipostorder)])
```

Comment:
This was not really testing much as n.iter_section calls core.iter_section. So if core.iter_section was bugged, the test would still pass.